src/test/shared/clients/defaultLambdaClient.test.ts

```diff
@@ -0,0 +1,98 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DefaultLambdaClient, LambdaClient } from '../../../shared/clients/lambdaClient'
+import { stub, createStubInstance, SinonStubbedInstance } from 'sinon'
+import { Lambda, Request } from 'aws-sdk'
+import * as assert from 'assert'
+import { FunctionUrlConfigList, FunctionUrlConfig, ListFunctionUrlConfigsResponse } from 'aws-sdk/clients/lambda'
+
+describe('Test the default lambda client', async () => {
+    let client: LambdaClient
+    const regionCode: string = 'my-region-code'
+    const lambdaFuncName = 'lambda-func-name'
+    let lambdaApiMock: SinonStubbedInstance<Lambda>
+
+    beforeEach(async () => {
+        client = new DefaultLambdaClient(regionCode)
+        lambdaApiMock = stub(new Lambda())
+
+        // '<any>' stops linter warning since method is private
+        stub(client, <any>'createSdkClient').resolves(lambdaApiMock)
```

Comment:
Stubbing a private method is usually a bad idea. We should think of ways to test our implementation without needing to know every little detail of the implementation.

Reply:
I've update the `createSdkClient()` in a separate commit. It now returns a singleton instance. Also you can pass in your own implementation to the class constructor, this will allow us to inject the mock instead of stubbing it as I did here.